### PR TITLE
Add search option

### PR DESCRIPTION
**Pull request by cp-luzi-p**

### CODE REVIEW

**Comment by cp-luzi-p** [comment]
Peek 2022-04-09 09-29

File: countrypicker/src/main/java/com/canopas/campose/countrypicker/CountrySearchView.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun countrySearchView(state: ModalBottomSheetState): String {
+
+    var searchValue: String by rememberSaveable { mutableStateOf("") }
+    var showClearIcon by rememberSaveable { mutableStateOf(false) }
+
+    if (searchValue.isEmpty()) {
+        showClearIcon = false
+    } else if (searchValue.isNotEmpty()) {
```
Comment (cp-radhika-s) [inline countrypicker/src/main/java/com/canopas/campose/countrypicker/CountrySearchView.kt]: `if (searchValue.isNotEmpty()` remove

File: countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt [hunk 2]
```diff
@@ -33,11 +34,13 @@ fun CountryPickerBottomSheet(
     show: Boolean,
     onItemSelected: (country: Country) -> Unit,
     onDismissRequest: () -> Unit,
+    dialogSearch: Boolean = true,
```
Comment (cp-radhika-s) [inline countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt]: What is use of `dialogSearch ` flag?

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

**Review by cp-hardik-p — COMMENTED** [review]
Can we use string resources here?

File: app/src/main/java/com/canopas/campose/jetcountypicker/MainActivity.kt [hunk 3]
```diff
@@ -44,29 +47,37 @@ fun SampleCountryPicker() {
...
+                expanded = false
+            },
+            onItemSelected = {
+                selectedCountry = it
+                expanded = false
+                focusManager.clearFocus()
+            }
+        ) {
             CountryTextField(
                 label = "Select country",
```
Comment (cp-hardik-p) [inline app/src/main/java/com/canopas/campose/jetcountypicker/MainActivity.kt]: Can we use string resources here?

**Comment by cp-luzi-p** [comment]
> Can we use string resources here?

yes

File: countrypicker/src/main/java/com/canopas/campose/countrypicker/CountrySearchView.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                colors = TextFieldDefaults.textFieldColors(
+                    backgroundColor = Color.Transparent,
+                    focusedIndicatorColor = Color.Transparent,
+                    disabledIndicatorColor = Color.Transparent,
+                    unfocusedIndicatorColor = Color.Transparent
+                )
+            )
+            if (searchValue.isEmpty()) {
+                Text(
+                    text = "Search",
```
Comment (cp-hardik-p) [inline countrypicker/src/main/java/com/canopas/campose/countrypicker/CountrySearchView.kt]: Can we use string resources here?

File: countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt [hunk 5]
```diff
@@ -60,32 +62,43 @@ fun CountryPickerBottomSheet(
...
+                            .clickable {
+                                selectedCountry = country
+                                onItemSelected(selectedCountry)
+                            }
+                            .padding(10.dp)) {
+                            Text(text = localeToEmoji(country.code))
+                            Text(
+                                text = country.name,
+                                modifier = Modifier
+                                    .padding(start = 6.dp)
```
Comment (cp-hardik-p) [inline countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt]: Add padding in multiples of 4

File: countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt [hunk 6]
```diff
@@ -60,32 +62,43 @@ fun CountryPickerBottomSheet(
...
+                                    .weight(2f)
+                            )
+                            Text(
+                                text = country.dial_code,
+                                modifier = Modifier
+                                    .padding(start = 6.dp)
```
Comment (cp-hardik-p) [inline countrypicker/src/main/java/com/canopas/campose/countrypicker/CountryPickerBottomSheet.kt]: Same as above

**Review by cp-megh-l — APPROVED** [review]

**Review by cp-hardik-p — APPROVED** [review]

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**